### ISSUE TITLE
API incompatibility between target frameworks

### ISSUE THREAD

**Issue by huoyaoyuan** [microsoft/CopyOnWrite#50]
The definition of `CloneFileAsync` is different for target frameworks:

https://github.com/microsoft/CopyOnWrite/blob/945d3160e764e2d76ad70656044aad315cb817ec/lib/ICopyOnWriteFilesystem.cs#L122-L129

The NuGet mechanism requires the API for higher framework being **strict superset** of lower framework. See more at https://learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/compatible-framework-in-package-validator .

In short, the assembly used for run time will be picked by the framework of *entry project*, but for compile time it will be picked by *current project*. Such incompatibility will cause `MissingMethodException` when using in a transitive project targeting netstandard2.0.
